client/src/views/workspaces/WorkspacesPage.vue

```diff
@@ -459,15 +483,17 @@ async function refreshWorkspacesList(): Promise<void> {
 }
 
 const filteredWorkspaces = computed(() => {
-  const filter = filterWorkspaceName.value.toLocaleLowerCase();
+  const filter = searchFilterContent.value.toLocaleLowerCase();
   return Array.from(workspaceList.value)
-    .filter((workspace) => workspace.currentName.toLocaleLowerCase().includes(filter))
+    .filter((workspace) => workspace.currentName.toLocaleLowerCase().includes(filter) && !isWorkspaceFiltered(workspace.currentSelfRole))
     .sort((a: WorkspaceInfo, b: WorkspaceInfo) => {
       if (favorites.value.includes(b.id) !== favorites.value.includes(a.id)) {
         return favorites.value.includes(b.id) ? 1 : -1;
       }
       if (sortBy.value === SortWorkspaceBy.Name) {
         return sortByAsc.value ? a.currentName.localeCompare(b.currentName) : b.currentName.localeCompare(a.currentName);
+      } else if (sortBy.value === SortWorkspaceBy.Role) {
+        return compareWorkspaceRoles(b.currentSelfRole, a.currentSelfRole) * (sortByAsc.value ? 1 : -1);
```

Comment:
It looks like roles are sorted like this:
- Ascending: `Owner` (first), `Manager`, `Contributor`, `Reader` (last)
- Descending: `Reader` (first), `Contributor`, `Manager`, `Owner` (last)

Shouldn't it be the opposite?